Gems/Camera/Code/Source/CameraComponentController.cpp

```diff
@@ -423,30 +475,37 @@ namespace Camera
             m_view->SetCurrentParams(viewParams);
         }
 
-        auto atomViewportRequests = AZ::Interface<AZ::RPI::ViewportContextRequestsInterface>::Get();
-        if (m_atomCamera && atomViewportRequests)
+        if (auto viewportContext = GetViewportContext())
         {
             AZ::Matrix4x4 viewToClipMatrix;
             float aspectRatio = m_view ? m_view->GetCamera().GetPixelAspectRatio() : 1.f;
-            auto viewportContext = atomViewportRequests->GetViewportContextByName(
-                atomViewportRequests->GetDefaultViewportContextName());
-            if (viewportContext)
+            if (!m_atomAuxGeom)
             {
-                if (!m_atomAuxGeom)
-                {
-                    SetupAtomAuxGeom(viewportContext);
-                }
-                auto windowSize = viewportContext->GetViewportSize();
-                aspectRatio = aznumeric_cast<float>(windowSize.m_width) / aznumeric_cast<float>(windowSize.m_height);
+                SetupAtomAuxGeom(viewportContext);
             }
+            auto windowSize = viewportContext->GetViewportSize();
+            aspectRatio = aznumeric_cast<float>(windowSize.m_width) / aznumeric_cast<float>(windowSize.m_height);
 
             // This assumes a reversed depth buffer, in line with other LY Atom integration
-            AZ::MakePerspectiveFovMatrixRH(viewToClipMatrix,
-                AZ::DegToRad(m_config.m_fov),
-                aspectRatio,
-                m_config.m_nearClipDistance,
-                m_config.m_farClipDistance,
-                true);
+            if (m_config.m_orthographic)
+            {
+                AZ::MakeOrthographicMatrixRH(viewToClipMatrix,
+                    -m_config.m_orthographicHalfWidth,
+                    m_config.m_orthographicHalfWidth,
+                    -m_config.m_orthographicHalfWidth / aspectRatio,
+                    m_config.m_orthographicHalfWidth / aspectRatio,
+                    m_config.m_farClipDistance,
+                    m_config.m_nearClipDistance);
+            }
+            else
+            {
+                AZ::MakePerspectiveFovMatrixRH(viewToClipMatrix,
+                    AZ::DegToRad(m_config.m_fov),
+                    aspectRatio,
+                    m_config.m_nearClipDistance,
+                    m_config.m_farClipDistance,
+                    true);
+            }
```

Comment:
question: just double checking the near/far arguments passed are correct. They have different order, in `MakeOrthographicMatrixRH` far is passed first, then near, and in `MakePerspectiveFovMatrixRH` near is passed first, then far.
Is that correct? If it is then we should change the functions to be consistent and avoid potential errors in the future.

Reply:
Ah yeah `MakeOrthographicMatrixRH` didn't have a `reverseDepth` parameter like the other helpers - I went ahead and added one to make this less confusing, as it's very easy to get wrong (including in the gif recorded above!)

Reply:
That's great, thanks!